lc2fen/infer_pieces.py

```diff
@@ -288,39 +288,39 @@ def __generate_fen_based_on_previous_fen_and_detected_move(
 def __determine_most_probable_white_piece(pieces_probs, square):
     """Determines the most probable white piece."""
     most_probable_piece_prob = 0
-    if pieces_probs[square][4] > most_probable_piece_prob:
+    if pieces_probs[square][__PREDS_DICT["Q"]] > most_probable_piece_prob:
         most_probable_piece = "Q"
-        most_probable_piece_prob = pieces_probs[square][4]
-    if pieces_probs[square][2] > most_probable_piece_prob:
+        most_probable_piece_prob = pieces_probs[square][__PREDS_DICT["Q"]]
+    if pieces_probs[square][__PREDS_DICT["N"]] > most_probable_piece_prob:
         most_probable_piece = "N"
-        most_probable_piece_prob = pieces_probs[square][2]
-    if pieces_probs[square][5] > most_probable_piece_prob:
+        most_probable_piece_prob = pieces_probs[square][__PREDS_DICT["N"]]
+    if pieces_probs[square][__PREDS_DICT["R"]] > most_probable_piece_prob:
         most_probable_piece = "R"
-        most_probable_piece_prob = pieces_probs[square][5]
-    if pieces_probs[square][0] > most_probable_piece_prob:
+        most_probable_piece_prob = pieces_probs[square][__PREDS_DICT["R"]]
+    if pieces_probs[square][__PREDS_DICT["B"]] > most_probable_piece_prob:
         most_probable_piece = "B"
-        most_probable_piece_prob = pieces_probs[square][0]
-    if pieces_probs[square][3] > most_probable_piece_prob:
+        most_probable_piece_prob = pieces_probs[square][__PREDS_DICT["B"]]
+    if pieces_probs[square][__PREDS_DICT["P"]] > most_probable_piece_prob:
         most_probable_piece = "P"
     return most_probable_piece
 
 
 def __determine_most_probable_black_piece(pieces_probs, square):
     """Determines the most probable black piece."""
     most_probable_piece_prob = 0
-    if pieces_probs[square][11] > most_probable_piece_prob:
+    if pieces_probs[square][__PREDS_DICT["q"]] > most_probable_piece_prob:
         most_probable_piece = "q"
-        most_probable_piece_prob = pieces_probs[square][4]
```

Comment:
Here I see that the initial code doesn't correspond to the black pieces but to the white pieces. I think these changes are correct, but @David-davidlxl can you confirm that the new behavior with these changes is the intended one?

Reply:
Correct! Sorry for the initial implementation error that I failed to catch